File: gto-support-db/src/testCommon/java/org/ccci/gto/android/common/db/CompoundMapper.java
package org.ccci.gto.android.common.db;

import android.content.ContentValues;
import android.database.Cursor;
import androidx.annotation.NonNull;

import org.ccci.gto.android.common.db.model.Compound;

import static org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_DATA1;
import static org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_DATA2;
import static org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_ID1;
import static org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_ID2;

class CompoundMapper extends AbstractMapper<Compound> {
    @Override
    protected void mapField(@NonNull ContentValues values, @NonNull String field, @NonNull Compound obj) {
        switch (field) {
            case COLUMN_ID1:
                values.put(field, obj.id1);
                break;
            case COLUMN_ID2:
                values.put(field, obj.id2);
                break;
            case COLUMN_DATA1:
                values.put(field, obj.data1);
                break;
            case COLUMN_DATA2:
                values.put(field, obj.data2);
                break;
            default:
                super.mapField(values, field, obj);
        }
    }

    @NonNull
    @Override
    protected Compound newObject(@NonNull Cursor c) {
        return new Compound(getString(c, COLUMN_ID1, ""), getString(c, COLUMN_ID2, ""));
    }

    @NonNull
    @Override
    public Compound toObject(@NonNull Cursor c) {
        final Compound obj = super.toObject(c);
        obj.data1 = getString(c, COLUMN_DATA1);
        obj.data2 = getString(c, COLUMN_DATA2);
        return obj;
    }
}

File: gto-support-db/src/androidTest/java/org/ccci/gto/android/common/db/CompoundMapper.kt
package org.ccci.gto.android.common.db

import android.content.ContentValues
import android.database.Cursor
import org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_DATA1
import org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_DATA2
import org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_ID1
import org.ccci.gto.android.common.db.Contract.CompoundTable.COLUMN_ID2
import org.ccci.gto.android.common.db.model.Compound
import org.ccci.gto.android.common.util.database.getString

internal object CompoundMapper : AbstractMapper<Compound>() {
    override fun mapField(values: ContentValues, field: String, obj: Compound) {
        when (field) {
            COLUMN_ID1 -> values.put(field, obj.id1)
            COLUMN_ID2 -> values.put(field, obj.id2)
            COLUMN_DATA1 -> values.put(field, obj.data1)
            COLUMN_DATA2 -> values.put(field, obj.data2)
            else -> super.mapField(values, field, obj)
        }
    }

    override fun newObject(c: Cursor) = Compound(
        c.getString(COLUMN_ID1, ""),
        c.getString(COLUMN_ID2, "")
    )

    override fun toObject(c: Cursor) = super.toObject(c).apply {
        data1 = c.getString(COLUMN_DATA1)
        data2 = c.getString(COLUMN_DATA2)
    }
}
